### DISCUSSION TITLE
Dropdown menu open by default

### DISCUSSION THREAD

**Discussion by iammuttaqi** [htmlstreamofficial/preline#269] [Q&A]
![image](https://github.com/htmlstreamofficial/preline/assets/44245907/112aa579-d6b0-42dd-9369-6487efd2f53f)

any way to open this dropdown menu by default?

**Comment by biplab-sutradhar** [comment]
If you are using it in react toggle the boolean state and for js change display : none to display : block

**Comment by fsmeier** [comment]
i also looked for this but can not get it working. or just partly: when i do sth like lg:hidden it shows on mobile without clicking it but the then non-mobile version does not show anymore :/

**Comment by langscot** [comment]
You won't be able to have it open by default properly with just changing classes. This is because the Dropdown plugin uses Popper behind the scenes, and when opened/closed it creates a popper instance to have the placement work correctly etc.

Your best bet is likely using the `HSDropdown`'s `open()` methods, either on the instance or the static one, whatever fits best for your code.

You could open a feature request issue so that `HSDropdown` will call `open()` when initialized if it sees the `open` class in the class list for example. That would be nice.
